### PR TITLE
EnC: simplify SymbolMatcher

### PR DESCRIPTION
**Pull request by tmat**
Implements https://github.com/dotnet/roslyn/issues/51595

### CODE REVIEW

**Comment by tmat** [comment]
@davidwengier @dotnet/roslyn-compiler @cston PTAL

File: src/Compilers/Core/Portable/Symbols/ISymbolInternal.cs [hunk 1]
```diff
@@ -27,6 +27,8 @@ internal interface ISymbolInternal
 
         string MetadataName { get; }
 
+        int MetadataToken { get; }
```
Comment (jjonescz) [inline src/Compilers/Core/Portable/Symbols/ISymbolInternal.cs:30]: Why is this new declaration needed?

**Review by jjonescz — APPROVED** [review]

File: src/Compilers/Core/Portable/Symbols/ISymbolInternal.cs (hunk 1 above)
Comment (cston) [inline src/Compilers/Core/Portable/Symbols/ISymbolInternal.cs:30]: >MetadataToken [](http://example.com/codeflow?start=12&length=13)

Consider adding the comment from `ISymbol.MetadataToken`.

**Review by cston — APPROVED** [review]

File: src/Compilers/Core/Portable/Symbols/ISymbolInternal.cs (hunk 1 above)
Comment (tmat) [reply src/Compilers/Core/Portable/Symbols/ISymbolInternal.cs:30]: Used in DefinitionMap.GetInitialMetadataHandle

**Pull request merged**